lib/bridge.js

```diff
@@ -237,11 +240,109 @@ Bridge.prototype.run = function(port, config, appServiceInstance) {
         }
         self.opts.controller.onLog(line, false);
     });
+    this.customiseAppservice(this.appService);
     this.appService.listen(port);
     return this.loadDatabases();
 };
 
 /**
+ * Apply any customisations required on the appService object.
+ * Subclasses are intended to override this method where appropriate. If
+ * overridden the subclass method will still have to call the one in this
+ * class.
```

Comment:
Whilst the `Bridge` class *supports* subclassing, I dislike *advocating* it. There's a number of reasons for this:
 - You need to remember to call the parent method, as you mention here. This is easily missed.
 - It's not clear what can be subclassed and what cannot, nor is it clear *in the subclass* what is an overridden method. There is no visibility keyword or `@Override` annotations.
 - You ultimately need to know the inner workings of the parent method in order to fully utilise child methods. It's not clear what happens in `super.customiseAppservice` at all without actually digging through the code.
 - Javascript in particular has no real concept of inheritance, just prototypes, so trying to emulate inheritance feels a bit weird (though to be fair, ES6 are actively attempting to emulate it :disappointed: ).

From looking at the code, it looks like your **intention** is to allow `Bridge` subclasses the ability to add new paths to the `appService` itself. Why not expose that functionality instead e.g:

```javascript
// handlerFn = function(req, res){}
Bridge.prototype.addAppServicePath = function(method, path, handlerFn) {
    this.appService.app[method.toLowerCase()](path, handlerFn);
};
```

Express is [a little bit gross](http://expressjs.com/en/api.html#app.METHOD) with how you select the methods, but \*shrug\*. This makes it much clearer what "customisation" is being done, and keeps the library in control of *how* this handler function gets added to the `AppService` object in case we want to change it later down the line.